Tests/AnalyticsTests/Tests.swift

```diff
@@ -18,9 +18,30 @@ final class Tests: XCTestCase {
         XCTAssertEqual(observer.eventName, ViewEvent.view.name)
         XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
     }
+    
+    func testLogEventAction() {
+        var values = AnalyticsValues()
+        values[keyPath: \.source] = .contactList
+        var action = AnalyticsAction(values: values)
+
+        action(.view)
+        XCTAssertEqual(observer.eventName, ViewEvent.view.name)
+        XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
+    }
+    
+    func testLogEventReplacing() {
+        var values = AnalyticsValues()
+        values[keyPath: \.source] = .contactList
+        var action = AnalyticsAction(values: values)
+
+        action(.view, appending: values)
```

Comment:
Can remove this since it doesn't add any value to the test. If you added assertions after this line to check that 'normal' logs would indeed have the `source` value perhaps, but I'm not sure that's super valuable here.

Rather we should test:
- that the values are _not empty_
- calling the `replacing` method has empty values (as you've done)
- then perhaps a normal log to assert that the original `source` value hasn't been removed from subsequent calls.

 you should add an assertion to verify that 
```suggestion
XCTAssertFalse(values.params.isEmpty)
```